com.unity.render-pipelines.universal/Editor/Camera/UniversalRenderPipelineCameraEditor.cs

```diff
@@ -259,6 +259,12 @@ public new void OnDisable()
             base.OnDisable();
         }
 
+        // IsPreset is an internal API - lets reuse the usable part of this function
+        static internal bool IsPresetEditor(UnityEditor.Editor editor)
+        {
+            return (int)((editor.target as Component).gameObject.hideFlags) == 93;
```

Comment:
same missing null check?

Reply:
There is no HideFlags for this yet. so i need to cast to int because we go from Enum => int comparison.
When the API will be public this will not be necessary anymore

Reply:
You could also cast 64 to the enum until the missing entry has been exposed. This will allow you to construct the combination of flags used here and compare using the enum, like [this](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgZgAIMiBhIgbzyNqIG0AxAGwEMBzAZwF0a7CiAUygQAtkQASASziCWHTn1rVcdNUQD0GogEEiUAPZhRrZjCIA3KZykAjZoKIHbAK0EBjYADoiAFQAW1kRBwP6OsgBmrBDM3krqAHIGUI4AvEQADDB48WpafmFOrh7ARADuUszM+galrAAO9YKsYMFQRKGOgYJgLe7+AJ5ErFBw5ZXVhqWc/gZlRBD1bR2F9WAGbp6WUoLzUhHBpUGcwEaCY1LtI8OcnIJxquq00rIAklDSPX2DROmY5locLhcnR8q8jpwaqV6gZbnYHB0DNtdodlp02pwmp4jCDnjJBO93piSkZfkR0NlgY8nvkAo5nJtShUqlCiLZHJxWBZzoiVhz3MJBF5cUQACLJYAAZS5BKgAFE4FJTq10gAWAEaACsOWp6lphQZJWCkKmbMc5yVrHsjkufIxWOVIqSwAVlutZIAHDB8pgCDqnrR9fTiltmZNamaiJzuWMymF2rYIJVFVB2MMiPU2AMeiLxVApTL3gAhJPMMZ/ABs3o06AAnP6A0GiozxizTeyFlBmAZWLIxrYhgBVLs9uDDiB3OA6W73RS6tR54DD7u98eT6d3UrpAgUhtPRfS7lk1VERBETAV09EHeUkX5PTuAyiWyXVjAKTJJwHU0zObtW3ot0vRgP0QwjGMP4ymMpy8pwAopJwwrznQLyCDoowHjKZKYFeJ5nh6V4XleO7xAAvnuRDxBg6CUbqKgBpYLQdC07D3AA8iGwCofIXBkgAFNxbBcAAlLWBAANwUWoFhMdYAAKYCCJugkKGSKlcF4qHvJ8wGgbRDEGeoyCSPiPGIZh3LvK6jrIYZTzGepiHOtZVoOCKdl0A5plCeZEqHrKJbJu5HlEMZAneQowkVqqklUiKqCYLWfEACQAESvJwClKfcIBUMALHsZxjm/Ok8mKcpEVcKRqXCbF8WJSl6WZeVOVUI5XgWWkpXNdlXGVZw1W1bid4aAlSX5WArHABxjLFQAZCZshmV4SQpENtmaKNDUTVNM0lPNi1yD5mn4tpOy6YM60MVoY18TthWzf1RALe1WlQESDpGFdjZbeNBXTUVT0vf1HV+YW8qKsq300r9d3/XtngHe1zmQ65gjQ3qsP3QDj1LT5z2HctnXFqWcAY3kWPw4DeOqcDNMaYuy6jmu5wbvc5OgrDZDJNyYDeL4BiSsAYCXOwfF8ZcwDCdjCN9fTnDmOgwkY+RuCkUAA=).

Reply:
i really don't understand the benefits of your proposal, the code I have could be searched for and replaced once the API is made public. maybe @RSlysz you could find the right words to explain to me what i fail to understand?
@svens-unity Regarding the rest of the PR, is there anything blocking?

Reply:
Honestly I dunno if it will ever made public. Preset are such an internal things.

I think what @svens-unity propose is just to be extra explicit here:
```suggestion
            const HideFlags preset = 64 //usefull comment about why HideFlags 64 is not exposed here
                | HideFlags.DontSaveInEditor
                | HideFlags.NotEditable
                | HideFlags.DontSaveInBuild;
            return (editor.target as Component).gameObject.hideFlags == preset;
```
This way we don't lost information like "this is not saved".

Reply:
It'd improve code readability. The check here in `IsPresetEditor` grabs the GameObject of the targeted component. Then it checks if exactly all of the following flags are enabled:
- HideFlags.HideInHierarchy 
- HideFlags.DontSaveInEditor
- HideFlags.NotEditable
- HideFlags.DontSaveInBuild
- (HideFlags)64;

The suggested change makes this much more explicit. Anyone reading the code will understand it from reading this code right here. Understanding the `93` on the other hand would require the dev to convert it to bits and check in source code what flags are enabled.

Reply:
As an aside the access modifier and static keyword are in an incorrect order. Also this code could break if a new element is added to HideFlags that is always enabled. 

I did another pass 30 minutes ago.

Reply:
93 is not a combination of the flags actually but a magic number (asked Bastien, Preset dev about it). I have added the explanation as a comment for the future. Let me know if this is better?

Reply:
[HideFlags](https://github.cds.internal.unity3d.com/unity/unity/blob/22f5c73389b2d2d6a06e7d4a795d1757b0198ffa/Runtime/Export/Scripting/UnityEngineObject.bindings.cs#L11) use the [Flags attribute](https://docs.microsoft.com/en-us/dotnet/api/system.flagsattribute?view=net-5.0), each individual flag can be enabled or disabled. In this code you're checking if the `int` representation of those flags is equal to 93. In bits `93` is equal to `1011101`, in other words the 1st, 3rd, 4th, 5th and 7th flag are enabled, all other flags are disabled. Therefore 93 is a combination of the flags I mentioned before.

Hence the other concerns I raised before. If someone introduces a new flag that is always enabled the value jumps from 93 to 221. Furthermore, this combination of flags might not be exclusive to presets? A user or any unity dev could create a GameObject with these HideFlags on a GameObject.